lib/buffer.js

```diff
@@ -272,19 +272,66 @@ Buffer.concat = function(list, length) {
 };
 
 
+function base64ByteLength(str) {
+  var len = str.length;
+  var bytes = len;
+
+  // Handle padding
+  if (str[len - 1] === '=')
+    bytes--;
+  if (len > 2 && str[len - 2] === '=')
+    bytes--;
+
+  // Base64 ratio: 3/4
+  bytes = (bytes / 4) * 3;
```

Comment:
It looks like they all have about the same speed, at least for me. Should I optimize that anyways? http://jsperf.com/base64-decoding-fancyness